### PR TITLE
The backend part of Comment, needs review.

### PR DESCRIPTION
**Pull request by 7bw**

### CODE REVIEW

File: backend/presentation/Viewsets/comment_view.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+'''
+URL: ://service/author/{author_id}/posts/{post_id}/comments access
+GET get comments of the post
+POST if you post an object of “type”:”comment”, it will add your comment to the post
+'''
+
+'''
+Manual Test:
+POST:
+{"displayName": "Lara Croft","github": "http://github.com/laracroft","username":"LaraCroft","email": "lara@gmail.com","password": "lara1234"}
```
Comment (Xuechunqiu) [inline backend/presentation/Viewsets/comment_view.py:18]: You can delete this or replace it with your test cases

Comment (xqiu1) [inline backend/presentation/Viewsets/comment_view.py]: You can delete this or replace it with your test cases

File: backend/presentation/Viewsets/comment_view.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    return post_id
+
+def getCommentIDFromRequestURL(request, id):
+    comment_id = f"/comments/{id}"
+    return comment_id
+
+class CommentViewSet(viewsets.ModelViewSet):
+    serializer_class = CommentSerializer
+    queryset = Comment.objects.all()
+    lookup_field = 'id'
```
Comment (xqiu1) [inline backend/presentation/Viewsets/comment_view.py]: No needed actually 😅  I forgot to delete it.
```suggestion
```

File: backend/presentation/Viewsets/comment_view.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        post_id = author_id + post_id
+        comment_id = post_id + comment_id
+        post = get_object_or_404(Post, id=post_id)
+        comments = get_object_or_404(Comment, post=post)
+        if comment_id in comments.items:
+            # Is there a problem about the response?
+            f = get_object_or_404(Comment, id=comment_id)
+            return Response({'exist': True})
+        else:
+            return Response({'exist': False}, 404)
```
Comment (xqiu1) [inline backend/presentation/Viewsets/comment_view.py]: This function is used to check if a comment exists, I think this one is not required.

File: backend/presentation/Viewsets/comment_view.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        comments = get_object_or_404(Comment, post=post)
+        comment_id = getCommentIDFromRequestURL(
+            request, self.kwargs['comment_id'])
+        comment_id = post_id + comment_id
+        # Possible mistake?
+        try:
+            comments.items.remove(comment_id)
+            followers.save()
+        except ValueError:
+            return Response("No such a follower. Deletion fails.", 500)
```
Comment (xqiu1) [inline backend/presentation/Viewsets/comment_view.py]:
```suggestion
            return Response("No such a comment. Deletion fails.", 500)
```

File: backend/socialdist/urls.py [hunk 5]
```diff
@@ -28,6 +28,10 @@
          FollowerViewSet.as_view({'get': 'list'})),
     path('author/<str:author_id>/followers/<str:foreign_author_id>/',
          FollowerViewSet.as_view({'get': 'retrieve'})),
+    path('author/<str:author_id>/posts/<str:post_id>/comments',
+        CommentViewSet.as_view({'get': 'list'})),
+    path('author/<str:author_id>/posts/<str:post_id>/comments/<str:comment_id',
+        CommentViewSet.as_view({'get': 'retrieve'})),
```
Comment (xqiu1) [inline backend/socialdist/urls.py:40]: If you name a function different from the HTTP method, you should define it within `as_view({})`.

**Review by xqiu1 — CHANGES_REQUESTED** [review]
You can use Postman to send a request to the server so that you can know if your methods work correctly.

File: backend/presentation/Serializers/request_serializer.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+from rest_framework import serializers
+from presentation.models import Request
+
+
+class RequestSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Request
+        fields = ['_type', 'summary', 'actor', 'object']
```
Comment (xqiu1) [inline backend/presentation/Serializers/request_serializer.py]:
```suggestion
        fields = ['type', 'summary', 'actor', 'object']
```

**Review by xqiu1 — CHANGES_REQUESTED** [review]

**Review by xqiu1 — APPROVED** [review]
LGTM

**Pull request merged**